application/common/components/Emailer.php

```diff
@@ -776,21 +776,38 @@ public function sendPasswordExpiringEmails()
             'status' => 'starting',
         ];
 
-        $users = User::getActiveUnlockedUsers();
+        $users = \Yii::$app->getDb()->createCommand("SELECT u.*
```

Comment:
Another option for reducing memory usage would be to use Yii2's built in batching capability. Docs: https://www.yiiframework.com/doc/guide/2.0/en/db-active-record#data-in-batches

Reply:
That could also help us keep the query logic in the User class, rather than putting it here in the Emailer component.

Reply:
I considered moving it to the User class. That class is already very bloated, so I resisted the urge.

Reply:
> Another option for reducing memory usage would be to use Yii2's built in batching capability. Docs: https://www.yiiframework.com/doc/guide/2.0/en/db-active-record#data-in-batches

Thanks for this tip!